frontend/apps/picasso/stores/defi/polkadot/transfers/subscribers/feeItemEd.ts

```diff
@@ -12,33 +32,40 @@ export const subscribeFeeItemEd = async (api: ApiPromise) => {
     }),
     async ({ feeItem, isLoaded, sourceChain }) => {
       if (!isLoaded) return;
-      const assetId =
-        useStore.getState().substrateTokens.tokens[feeItem].chainId[
-          sourceChain
-        ];
 
-      if (!assetId) {
-        return;
+      function fetchEd(api: ApiPromise) {
+        return function (a: string | BigNumber) {
+          return TaskEither.tryCatch(
+            async () => await api.query.currencyFactory.assetEd(a.toString()),
+            (reason) => new Error(String(reason))
+          );
+        };
       }
 
-      try {
-        const ed = await api.query.currencyFactory.assetEd(assetId.toString());
-        const existentialString = ed.toString();
-        const existentialValue = fromChainIdUnit(
-          new BigNumber(existentialString)
-        );
-        useStore.setState({
-          ...useStore.getState(),
-          transfers: {
-            ...useStore.getState().transfers,
-            feeItemEd: existentialValue.isNaN()
-              ? new BigNumber(0)
-              : existentialValue,
-          },
-        });
-      } catch (e) {
-        console.log("Error while fetching existential deposit");
-      }
+      pipe(
+        Option.fromNullable(
+          useStore.getState().substrateTokens.tokens[feeItem].chainId[
+            sourceChain
+          ]
+        ),
+        Option.map(fetchEd(api)),
+        Option.map(
+          flow(
+            TaskEither.map(anyToBigNumber),
+            TaskEither.map((existentialValue) => {
+              useStore.setState({
```

Comment:
can we export a method from slice that updates state rather than updating the state here?